src/supplemental/nanolib/parquet/parquet.cc

```diff
@@ -490,6 +484,46 @@ parquet_write(
 			goto again;
 	}
 
+	char md5_buffer[33];
+	int ret = Compute_file_md5(filename, md5_buffer);
+	if (ret != 0) {
+		log_error("Failed to calculate md5sum");
+		free(filename);
+		return -1;
+	}
+
+	char *md5_file_name = (char *)malloc(strlen(filename) + strlen("_") + strlen(md5_buffer) + 2);
+	if (md5_file_name == NULL) {
+		log_error("Failed to allocate memory for file name.");
+		free(filename);
+		return -1;
+	}
+
+	strncpy(md5_file_name, filename, strlen(conf->dir) + strlen(conf->file_name_prefix) + 1);
+	md5_file_name[strlen(conf->dir) + strlen(conf->file_name_prefix) + 1] = '\0';
+
+	strcat(md5_file_name, "_");
+	strcat(md5_file_name, md5_buffer);
+	strcat(md5_file_name, filename + strlen(conf->dir) + strlen(conf->file_name_prefix) + 1);
+
+	ret = rename(filename, md5_file_name);
+	if (ret != 0) {
+		log_error("Failed to rename file %s to %s errno: %d", filename, md5_file_name, errno);
+		free(filename);
+		free(md5_file_name);
```

Comment:
Need to pthread_mutex_unlock(&parquet_queue_mutex);

Reply:
thx, done